src/snapred/backend/dao/indexing/Versioning.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Optional
+
+from numpy import integer
+from pydantic import BaseModel, computed_field, field_serializer
+from snapred.meta.Config import Config
+
+VERSION_START = Config["version.start"]
+VERSION_NONE_NAME = Config["version.friendlyName.error"]
+VERSION_DEFAULT_NAME = Config["version.friendlyName.default"]
+VERSION_DEFAULT = -1  # SNAPRed Internal flag for default version
```

Comment:
Unfortunately I had more thoughts on this.  Would it simplify things to regor it as such?

```
class VersionFlag(str, Enum):
    VERSION_START = Config["version.start"]
    VERSION_NONE_NAME = Config["version.friendlyName.error"]
    VERSION_DEFAULT_NAME = Config["version.friendlyName.default"]
    VERSION_DEFAULT = "-1"
    def __str__(self):
        return self.value

class Version:
    value: int | VersionFlag
    def __init__(self, value):
        if isinstance(value, int) and value >= VERSION_START:
            raise ValueError....

class VersionedObject(BaseModel):
    __version: Optional[Version] = None 
   
    ...
    
    def parseVersion(cls, version, *, exclude_none: bool = False) -> int | None:
       v = None
        if a in set(VersionFlag): 
            v = VersionFlag(v)
        else:
            v = int(v)

        if v == VersionFlag.VERSION_NONE_NAME and exclude_none:
            raise ValueError("Attempted to set version to None where not acceptable.")
   

    def writeVersion(cls, version) -> int | str:
        return parseVersion(v)
        
```

Reply:
I wrote up a toy example as well just to confirm some details:
```

# Online Python - IDE, Editor, Compiler, Interpreter

from enum import Enum 

class StrEnum(str, Enum):
    pass

class SomeFlag(StrEnum):
    One = "1"
    Two = "2"
    
    def __str__(self):
        return self.value

def parseA(a):
    if a in set(SomeFlag): 
        a = SomeFlag(a)
        return a
    else:
       return int(a) 

def assertFlag(a :SomeFlag):
    assert isinstance(a, SomeFlag)
    
assertFlag(parseA("1"))
assertFlag(SomeFlag.One)
assert str(SomeFlag.One) == "1"
assert SomeFlag.One == "1"
assert parseA(1) == 1

```